datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -656,10 +656,22 @@ impl LogicalPlan {
                         ref limit,
                         ..
                     }) => {
+                        let projected_fields = match projection {
+                            Some(indices) => {
+                                let schema = source.schema();
+                                let names: Vec<&str> = indices
+                                    .iter()
+                                    .map(|i| schema.field(*i).name().as_str())
+                                    .collect();
+                                format!("Some([{}])", names.join(", "))
```

Comment:
What about only showing the projection when there is one and ommiting it when there are none.
This could remove the None/Some too:

`TableScan a projection=[col1,col2] filter`
vs
`TableScan a`

Reply:
I started out doing that and it broke many tests that were expecting `None` so I decided to reduce the size of the change. I agree that it would be a nice improvement though. I'm happy to work on that as a follow-on PR.

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/2697